common/test/acceptance/tests/discussion/test_discussion.py

```diff
@@ -997,6 +997,28 @@ def test_mathjax_rendering_in_order(self):
 
         self.assertEqual(self.page.get_new_post_preview_text(), 'Text line 1\nText line 2')
 
+    def test_mathjax_not_rendered_after_post_cancel(self):
+        """
+        Tests that mathjax is not rendered when we cancel the post
+
+        When user types the mathjax expression into discussion editor, it will appear in te preview
+        box, and when user cancel it and again click the "Add new post" button, mathjax will not
+        appear in the preview box
+        """
+        self.page.fill_title_field("MathJax Title")
+        self.page.set_new_post_editor_value(
+            '\\begin{equation}'
+            '\\tau_g(\omega) = - \\frac{d}{d\omega}\phi(\omega) \hspace{2em} (1) '
+            '\\end{equation}'
+        )
+        self.page.preview_mathjax_visiblity()
+        self.page.click_element(".cancel")
+        alert = get_modal_alert(self.browser)
+        alert.accept()
+        self.assertIsNotNone(self.page.new_post_button)
+        self.page.click_new_post_button()
+        self.assertEqual(self.page.get_preview_value(), "")
```

Comment:
You may want to check element visibility here before retrieving ( `get_preview_value()`) . My suggestion is to use `get_new_post_preview_text` or `get_new_post_preview_value` if it is applicable.